### PR TITLE
Feature/upload file

### PR DESCRIPTION
**Pull request by tatianaIljina**
Base component for changing state of boolean property
Added QuickSearch for property of inner object

### CODE REVIEW

File: API/Tenders.API/Controllers/OpenCompetitionFileController.cs [hunk 1]
```diff
@@ -194,6 +194,32 @@ public FileContentResult SaveFile(string id)
...
 
+        /// <summary>
+        /// Makes file disabled.
+        /// </summary>
+        /// <param name="id">Open comoetition file id.</param>
+        /// <returns>Result.</returns>
+        [HttpPatch("SetValueOff")]
+        public ActionResult SetValueOff(string id)
+        {
+            var file = this.fileRepo.Find(id).FirstOrDefault();
```
Comment (Anny777) [inline API/Tenders.API/Controllers/OpenCompetitionFileController.cs]: Not enough check
If not found return 404
If not update return 409

File: API/Tenders.API/Controllers/OpenCompetitionFileController.cs [hunk 2]
```diff
@@ -194,6 +194,32 @@ public FileContentResult SaveFile(string id)
...
+
+        /// <summary>
+        /// Makes file disabled.
+        /// </summary>
+        /// <param name="id">Open comoetition file id.</param>
+        /// <returns>Result.</returns>
+        [HttpPatch("SetValueOn")]
+        public ActionResult SetValueOn(string id)
+        {
+            var file = this.fileRepo.Find(id).FirstOrDefault();
```
Comment (Anny777) [inline API/Tenders.API/Controllers/OpenCompetitionFileController.cs]: Same

File: Core/Tenders.Core/Services/QuickSearchService.cs [hunk 3]
```diff
@@ -78,6 +80,12 @@ public void GenerateQuickSearchString(IModelBase model)
                         var mb = (ModelBase)property.GetValue(model);
                         result = mb?.QuickSearch;
                     }
+                    else if (property.PropertyType.IsInterface && property.PropertyType.IsAssignableFrom(typeof(ICustomer)))
```
Comment (Anny777) [inline Core/Tenders.Core/Services/QuickSearchService.cs]: This is not necessary, please ask me why, I will explain :)

File: API/Tenders.API/Controllers/OpenCompetitionFileController.cs (hunk 1 above)
Comment (tatianaIljina) [reply API/Tenders.API/Controllers/OpenCompetitionFileController.cs]: 409 is conflict status code. Are you sure that it is suitable for this error?

File: tests/Tenders.API.Tests/OpenCompetitionFileControllerTest.cs [hunk 4]
```diff
@@ -207,6 +207,84 @@ public void SaveFileOk()
...
+        /// <summary>
+        /// Determines whether set value off returns Ok().
+        /// </summary>
+        [Fact]
+        public void SetValueOffOk()
+        {
+            this.fileRepo
+                .Setup(c => c.Update(It.IsAny<OpenCompetitionFile>()))
+                .Returns(true);
+            var result = this.sut.SetValueOff("ea9c1268-f286-4818-ab80-c670aab36d5a") as OkResult;
```
Comment (Anny777) [inline tests/Tenders.API.Tests/OpenCompetitionFileControllerTest.cs]: Assert.Type<OkResult>() not suitable?

File: API/Tenders.API/Controllers/OpenCompetitionFileController.cs [hunk 5]
```diff
@@ -194,6 +194,75 @@ public FileContentResult SaveFile(string id)
...
         }
 
+        /// <summary>
+        /// Makes file disabled.
+        /// </summary>
+        /// <param name="id">Open comoetition file id.</param>
+        /// <returns>Result.</returns>
+        /// /// <response code="404">Emails wasn't found.</response>
+        [HttpPatch("SetValueOff")]
+        [ProducesResponseType(404)]
```
Comment (Anny777) [inline API/Tenders.API/Controllers/OpenCompetitionFileController.cs]: Missing description of error codes

**Pull request merged**